controllers/machinehealthcheck_controller.go

```diff
@@ -256,12 +258,58 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for _, t := range unhealthy {
 		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
 
-		if annotations.IsPaused(cluster, t.Machine) {
-			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+		if m.Spec.RemediationTemplate != nil {
+			// If external remediation request already exists,
+			// return early
+			if r.externalRemediationRequestExists(ctx, m, t) {
+				logger.V(2).Info("External Remediation Request already exists", "machine", t.Machine.Name, "templateGVK", m.Spec.RemediationTemplate.GroupVersionKind())
+				return ctrl.Result{}, nil
+			}
+
+			cloneOwnerRef := &metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       t.Machine.Kind,
+				Name:       t.Machine.Name,
+				UID:        t.Machine.UID,
+			}
+
+			from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "error retrieving remediation template %q %q %q", m.Spec.RemediationTemplate.APIVersion, m.Spec.RemediationTemplate.Kind, m.Spec.RemediationTemplate.Name)
```

Comment:
This is the first time we are getting a Remediation template, so we should surface potential misconfiguration to the users as a condition on MHC.
For consistency, I suggest to add a condition named "ExternalRemediationTemplateAvailable", and mark it FALSE if we fail to retrieve the template.

However there is a potential risk of condition flickering if in one cluster there are at the same time more than one external remediation templates in use, some of them misconfigured and some no. Not sure this is a problem we should worry about

Reply:
Adding the new "ExternalRemediationTemplateAvailable" condition sounds good to me.

Reply:
I think it makes sense to have a condition to surface problems with the external remediation templates since we have a condition on line 309 below.

However, should we have a generic condition like `ExternalRemediationRequest` and mark the condition `false` with appropriate condition reasons for the other cases. For example, `TemplateNotAvailableReason`, `TemplateNotCreatedReason` etc...
I may be not understanding how conditions can be leveraged properly but my thought process is - are any of the errors within this `if m.Spec.RemediationTemplate != nil` block worth raising to the user?